### DISCUSSION TITLE
how to map the container port to a fixed port?

### DISCUSSION THREAD

**Discussion by 123liuziming** [testcontainers/testcontainers-go#2736] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/40c4eccf-b7a6-4571-bd10-686611b433c8)
Like `withFixedExposedPort` in java

**Comment by kiview** [comment, marked as answer]
This is deprecated in Java as well. You have to use the [modifier](https://golang.testcontainers.org/features/creating_container/#advanced-settings) feature to enable this, but it will break overall Testcontainers semantics.
